examples/chip-tool/commands/common/CHIPCommand.cpp

```diff
@@ -225,6 +225,11 @@ void CHIPCommand::MaybeTearDownStack()
     {
         ShutdownCommissioner(commissioner.first);
     }
+    mCommissioners.clear();
+    sICDClientStorage.Shutdown();
+    sCheckInHandler.Shutdown();
+    sGroupDataProvider.Finish();
+    DeviceControllerFactory::GetInstance().Shutdown();
```

Comment:
Does this handle shuttign down mDefaultStorage and the various other things that MaybeSetUpStack inits but are not explicitly shut down here?  Could use some comments, if so.

Reply:
I will absolutely add comments (short answer: Yes, the other "things" can be re-`Init`d safely!)